src/Auth0.ManagementApi.Client/ApiConnection.cs

```diff
@@ -125,6 +128,13 @@ public ApiConnection(string token, string baseUrl)
                 }
             }
 
+            // Add the diagnostics header, unless user explicitly opted out of it
+            if (!object.ReferenceEquals(diagnostics, DiagnosticsHeader.Suppress))
+                request.AddHeader("Auth0-Client", diagnostics.ToString());
+
+            // Add the user agent
+            request.AddHeader("User-Agent", ".NET/PCL");

```

Comment:
Maybe we need to split the DiagnosticsHeader in 2 parts:

 - The opt in which contains system info like OS, .NET version, .... This is a nice to have for troubleshooting but not a necessity (it's more important on iOS)
 - One thing I think we would always need is to send the current version of the SDK. The reason is that if we make a breaking change (which we obviously avoid) or if we detect a security issue with an SDK, we need to be able to reach out to all customers using the impacted version of the SDK